src/matrix/storage/idb/Transaction.ts

```diff
@@ -36,106 +36,111 @@ import {OperationStore} from "./stores/OperationStore";
 import {AccountDataStore} from "./stores/AccountDataStore";
 
 export class Transaction {
-    constructor(txn, allowedStoreNames, IDBKeyRange) {
+    private _txn: IDBTransaction;
+    private _allowedStoreNames: string[];
+    private _stores: { [storeName : string] : any };
+
+    constructor(txn: IDBTransaction, allowedStoreNames: string[], IDBKeyRange) {
```

Comment:
use the enum for the allowed store names? Also see earlier comment for type of IDBKeyRange